enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,1012 @@
+
+subgraph management-cluster
+    cluster-csi-snapshot-controller-operator
+end
+
+subgraph guest-cluster
+    cluster-csi-snapshot-controller-operator <-.->|sync| CSISnapshotController{{CSISnapshotController CR}}
+    cluster-csi-snapshot-controller-operator -.->|write| ClusterOperator{{ClusterOperator CR}}
+    cluster-csi-snapshot-controller-operator -.->|read| Nodes{{Nodes}}
+    cluster-csi-snapshot-controller-operator -.->|write| ValidatingWebhookConfiguration{{ValidatingWebhookConfiguration for snapshot CRs}}
+end
+
+```
+
+##### csi-snapshot-controller objects
+Here is list of objects to be able to run the final csi-snapshot-controller and its webhook.    
+
+Legend:
+* Files
+  * Source yaml file, typically in operator's manifests or assetes or openshift/api. 
+* Who - who creates the object (in Hypershift)
+  * CPO - control-plane-operator.
+  * CVO - cluster-version-operator. It can create objects only in the guest cluster.
+  * CSCO - cluster-csi-snapshot-controller-operator.
+* Where - where is the object created (in HyperShift).
+  * MC = management cluster.
+  * G = guest cluster.
+
+| File                                         | Who  | Where | Description                                                                              |
+|----------------------------------------------|------|-------|------------------------------------------------------------------------------------------|
+| csi_snapshot_controller_operator_01_crd.yaml | CVO  | G     | Operator CRD.                                                                            |
+| manifests/02_operator_cr.yaml                | CVO  | G     | Operator CR.                                                                             |
+| manifests/03_configmap.yaml                  | ?    | ?     | Operator’s config map, what is it good for? TODO: check if we can remove it.             |
+| manifests/03_service.yaml                    | CPO  | MC    | Metrics service for the operator, unused? TODO: check if we can remove it.               |
+| manifests/04_serviceaccount.yaml             | CVO  | G     | Operator’s ServiceAccount for RBACs in the guest cluster.                                |
+ | NEW: operator + operand ServiceAccounts      | CPO  | MC    | Operator's ServiceAccount in the management  cluster. Not sure if it's needed.           |
+| manifests/05_operand_rbac.yaml               | CVO  | G     | Few RBAC objects for the operand. We can run the operand with kubeadmin for now.         |
+| manifests/05_operator_rbac.yaml              | CVO  | G     | Operator RBAC, basically cluster-admin binding. No special RBAC is needed in HyperShift. |
+| manifests/05_user_rbac.yaml                  | CVO  | G     | Aggregate RBAC roles to add to storage-admin and basic-user to manage snapshots.         |
+| manifests/07_deployment.yaml                 | CPO  | MC    | Operator's Deployment.                                                                   |
```

Comment:
How are we going to keep these manifests synchronized between snapshot-operator (which is managed by CVO in other environments) and by CPO in hypershift environment?

Reply:
We have to do something similar for MicroShift, and there we have automation to extract the manifests from the release payload and add them to our repo to be customized. We're working toward a model where the operators that own the assets can be run in a one-time mode that produces the same content, so the customization can be known and tested "upstream" of MicroShift. https://github.com/openshift/microshift/blob/main/scripts/rebase.sh#L277

Reply:
That OLM directory looks something like what we're doing in MicroShift.

Reply:
I guess we will sync the yamls manually, at least in the beginning.